src/ext/coms.c

```diff
@@ -69,10 +69,14 @@ static uint32_t store_data(group_id_t group_id, const char *src, size_t size) {
     return 0;
 }
 
-static ddtrace_coms_stack_t *new_stack(void) {
+static ddtrace_coms_stack_t *new_stack(size_t min_size) {
+    size_t size = DDTRACE_COMS_STACK_SIZE;
+    while (min_size > size) {
```

Comment:
Can we add a check to make sure that we do not add more then, 5MB?

Reply:
Of course; I just didn't get to it before lunch and wanted to have something for you to test in the meantime.